### PR TITLE
[main][test] script prepare test mainnetfork

### PR DESCRIPTION
**Pull request by blackalpacaa**
## Description

- Create scripts for setting up test environment.

## Why?

- In order to set the test environment on mainnetfork network.

### CODE REVIEW

File: tests/helpers/time.ts [hunk 1]
```diff
@@ -37,6 +37,19 @@ export async function increaseTimestamp(duration: BigNumber) {
   await advanceBlock();
 }
 
+export async function advanceTimestampAndBlock(duration: BigNumber) {
+  if (duration.isNegative()) throw Error(`Cannot increase time by a negative amount (${duration})`);
+
+  await ethers.provider.send("evm_increaseTime", [duration.toNumber()]);
+
+  const blockToAdvance = duration.div(3).toNumber();
```
Comment (alpanaca) [inline tests/helpers/time.ts:45]: Please make 3 as a constant instead of hardcoding.

File: scripts/AdvanceTime.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import * as timeHelpers from "../tests/helpers/time";
+
+async function main() {
+  await timeHelpers.advanceTimestampAndBlock(timeHelpers.HOUR);
```
Comment (alpanaca) [inline scripts/AdvanceTime.ts:4]: Would we able to specify how many hours OR days when we run the script?

**Comment by alpanaca** [comment]
@blackalpacaa Can you create some document to demonstrate how we can

1. Set up Mainnet Fork dedicate node
2. How to run each task

This can be included in README.md or separate mainnetfork.md

File: tests/helpers/time.ts [hunk 3]
```diff
@@ -6,6 +6,8 @@ export const DAY = ethers.BigNumber.from(86400);
 export const WEEK = DAY.mul(7);
 export const YEAR = DAY.mul(365);
 
+const SEC_PER_BLOCK = 3;
```
Comment (alpanaca) [inline tests/helpers/time.ts:9]: Didn't we change this to some bigger number?

**Review by alpanaca — APPROVED** [review]

**Review by jr-alpaca — APPROVED** [review]

**Pull request merged**